### PR TITLE
Module-scoped imports

### PR DESCRIPTION
**Pull request by AlexanderSher**
- Make modules and subpackages available for __init__.py packages
- Scope imports to module
- Fix completion in tensorflow:
image

### CODE REVIEW

File: src/Analysis/Ast/Impl/Modules/PythonModule.cs [hunk 1]
```diff
@@ -330,9 +330,7 @@ internal PythonModule(ModuleCreationOptions creationOptions, IServiceContainer s
 
         public void Reset(string content) {
             lock (AnalysisLock) {
-                if (content != Content || content == null) {
-                    ContentState = State.None;
-                    Services.GetService<IPythonAnalyzer>().InvalidateAnalysis(this);
+                if (content != Content) {
```
Comment (MikhailArkhipov) [inline src/Analysis/Ast/Impl/Modules/PythonModule.cs]: I had an issue when analysis didn't start on `pip install` reload.

File: src/Analysis/Ast/Impl/Modules/PythonPackage.cs [hunk 2]
```diff
@@ -25,28 +27,54 @@ namespace Microsoft.Python.Analysis.Modules {
     /// Represents package with child modules. Typically
     /// used in scenarios such as 'import a.b.c'.
     /// </summary>
-    internal sealed class PythonPackage : PythonModule, IPythonPackage {
-        private readonly ConcurrentDictionary<string, IPythonModule> _childModules = new ConcurrentDictionary<string, IPythonModule>();
+    internal sealed class PythonVariableModule : IPythonModule, IEquatable<IPythonModule> {
```
Comment (MikhailArkhipov) [inline src/Analysis/Ast/Impl/Modules/PythonPackage.cs]: Didn't like the `package` name? :-)

File: src/Analysis/Ast/Impl/Modules/PythonModule.cs (hunk 1 above)
Comment (AlexanderSher) [reply src/Analysis/Ast/Impl/Modules/PythonModule.cs]: Merge issue

File: src/Analysis/Ast/Impl/Modules/PythonPackage.cs (hunk 2 above)
Comment (AlexanderSher) [reply src/Analysis/Ast/Impl/Modules/PythonPackage.cs]: It isn't a package anymore. I actually don't know how to call it better. Need to update the comment btw.

**Comment by jakebailey** [comment]
This PR didn't fix it for me:

image

The analysis never completes, and it completion does the same thing as in #636.

**Comment by jakebailey** [comment]
Hmm, eventually it completed and `tf.nn` shows the right thing, but I can see in the logs quite a few repeat analysis items, like `tensorflow.python.estimator.keras(Library)` 5 times.

**Comment by AlexanderSher** [comment]
Can you send me a full log?

**Pull request merged**